--- /usr/bin/archivemail~	2007-09-18 13:46:04.000000000 +0300
+++ /usr/bin/archivemail	2007-09-18 13:46:06.983211472 +0300
@@ -1251,7 +1252,11 @@
     cache = IdentityCache(mailbox_name)
 
     msg = original.next()
-    while (msg):
+    while msg is not None:
+        if not msg:
+            vprint("ignoring invalid message")
+            msg = original.next()
+            continue
         msg_size = sizeof_message(msg)
         stats.another_message(msg_size)
         vprint("processing message '%s'" % msg.get('Message-ID'))
